lib/libxdp/tests/test_link_detach.c

```diff
@@ -47,16 +47,26 @@ static int check_link_detach(int ifindex) {
 		goto out;
 	}
 
-
 	ret = bpf_object__load(obj_prog);
 	if (ret) {
 		ret = -errno;
 		fprintf(stderr, "Couldn't load object: %s\n", strerror(-ret));
 		goto out;
 	}
 
-	link = bpf_program__attach_xdp(prog, ifindex);
-	if (!link) {
+	prog_fd = bpf_program__fd(prog);
+	if (prog_fd < 0) {
+		ret = -errno;
+		fprintf(stderr, "Couldn't get prog fd: %s\n", strerror(-ret));
+		goto out;
+	}
+
+	if (mode == XDP_MODE_SKB)
+		xdp_flags = XDP_FLAGS_SKB_MODE;
+	
+
+	ret = bpf_xdp_attach(ifindex, prog_fd, xdp_flags, NULL);
```

Comment:
This uses the netlink attach mode, not bpf_link, so now you're no longer testing bpf_link detachment. You'll need to use `bpf_link_create()`...